server/tilesource/geospatial.py

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+#############################################################################
+#  Copyright Kitware Inc.
+#
+#  Licensed under the Apache License, Version 2.0 ( the "License" );
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#############################################################################
+
+import gdal
+import osr
+import mapnik
+from PIL import Image
+import math
+
+from .base import FileTileSource
+
+try:
+    import girder
+    from girder import logger
+    from .base import GirderTileSource
+except ImportError:
+    girder = None
+    import logging as logger
+    logger.getLogger().setLevel(logger.INFO)
+
+
+class MapnikTileSource(FileTileSource):
+    name = 'mapnik'
+
+    def __init__(self, path, **kwargs):
+        super(MapnikTileSource, self).__init__(path, **kwargs)
+        self.dataset = gdal.Open(self._getLargeImagePath())
+
+        self.sizeX = self.dataset.RasterXSize
+        self.sizeY = self.dataset.RasterYSize
+        self.tileWidth = 256
+        self.tileHeight = 256
+        # # Fair assumption
+        self.levels = 20
+
+        # Earth circumference for web mercator
+        self.circumference = 20037508.342789244
+
```

Comment:
Can you add a subclass method for `getMetadata` that returns the information the base tilesource has, plus 
- `geospatial`: a boolean set to True so consumers of this tile source know that it is geospatial.  Eventually this can be used to choose a better default viewer.
- `bounds`: the geospatial bounds as a dictionary which contains the max/min x/y coordinates and the coordinate system that they are in.

It might also be nice (if known) if the maximum resolution of the underlying image is known, so that the appropriate maximum `levels` could be more intelligently guessed at.